### ISSUE TITLE
Questions about converting range image to point cloud

### ISSUE THREAD

**Issue by sz3623** [ethz-asl/TULIP#4]
![2](https://github.com/user-attachments/assets/b00d9540-36c3-48f7-9648-731309d83ef9)

Hello, there are some problems in converting the upsampled range image to point cloud. 

The generated point cloud has many noise points in the center. The pixel values ​​at some locations of the image are very small (close to zero), such as the image boundary area. However, the pixel values ​​at these locations should be equal to 0. Therefore, noise appears after these locations are converted to point clouds. 

Do you also encounter this problem? Our training loss is close to 0.01. We speculate that the training has not converged. Setting thresholds and filtering is also a solution.

Thank you for your attention and look forward to your reply.

**Comment by binyang97** [comment]
> ![2](https://private-user-images.githubusercontent.com/48540066/365533757-b00d9540-36c3-48f7-9648-731309d83ef9.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwODg2MjgsIm5iZiI6MTcyNjA4ODMyOCwicGF0aCI6Ii80ODU0MDA2Ni8zNjU1MzM3NTctYjAwZDk1NDAtMzZjMy00OGY3LTk2NDgtNzMxMzA5ZDgzZWY5LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTExVDIwNTg0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3ZGZiMTVmYTMwYmNmYTBmNGJkOGFkOGQ0NjlmYmMwNzRhZTJiZjRlYTM5NDM5ZWEzMDlhYzg2Yzk1M2Q5YTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OJeSskhqmVJLLqKwMDNfSVhJeMbPUuCycFdch9IClJc)
> 
> Hello, there are some problems in converting the upsampled range image to point cloud.
> 
> The generated point cloud has many noise points in the center. The pixel values ​​at some locations of the image are very small (close to zero), such as the image boundary area. However, the pixel values ​​at these locations should be equal to 0. Therefore, noise appears after these locations are converted to point clouds.
> 
> Do you also encounter this problem? Our training loss is close to 0.01. We speculate that the training has not converged. Setting thresholds and filtering is also a solution.
> 
> Thank you for your attention and look forward to your reply.

Hi,

yes, I've also had the issue. That's mainly because the network tends to predict something even if the pixels are invalid (no projected points) in the ground truth. You can clamp the prediction with a minimum range, something like 3 or 2 meters, to filter out those noisy points. That would make the range image look much cleaner. Another option is to let the network learn the projection mask at the same time and for example, use binary cross entropy loss to optimize the network. The final output is then the combination of predicted mask and range image. I have not tried this yet but I think it is also a feasible solution. 

Best, Bin
